docs/examples/201/wvd-backplane-with-network-and-storage-and-monitoring/wvd-fileservices-privateendpoint-module.bicep

```diff
@@ -0,0 +1,58 @@
+param location string
+param subnetId string
+param vnetId string
+param storageAccountId string
+param privateEndpointName string = 'privateEndpoint${uniqueString(resourceGroup().name)}'
+param privateLinkConnectionName string = 'privateLink${uniqueString(resourceGroup().name)}'
+param privateDNSZoneName string = 'privatelink.file.core.windows.net'
+
+resource privateEndpoint 'Microsoft.Network/privateEndpoints@2020-06-01' = {
+  name: privateEndpointName
+  location: location
+  properties: {
+    subnet: {
+      id: subnetId
+    }
+    privateLinkServiceConnections: [
+      {
+        name: privateLinkConnectionName
+        properties: {
+          privateLinkServiceId: storageAccountId
+          groupIds: [
+            'file'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource privateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: privateDNSZoneName
+  location: 'global'
+}
+
+resource virtualNetworkLink 'Microsoft.Network/privateDnsZones/virtualNetworkLinks@2020-06-01' = {
+  name: '${privateDNSZone.name}/${privateDNSZone.name}-link'
```

Comment:
You can use the new child resource syntax options. Either nest the resource or use the `parent` property:

nesting-style:
```bicep
resource privateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
  name: privateDNSZoneName
  location: 'global'

  resource virtualNetworkLink 'virtualNetworkLinks' = {
    name: '${privateDNSZone.name}-link'
    ...
  }
}
```

parent-style:
```bicep
resource virtualNetworkLink 'Microsoft.Network/privateDnsZones/virtualNetworkLinks@2020-06-01' = {
  parent: privateDNSZone
  name: '${privateDNSZone.name}-link'
  ...
```